gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,17 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.findDeviceFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                n.deviceID.name = "Unknown";
+                return n;
+            }
+            return matches[0];
+        };
+
```

Comment:
Actually it should not be 'Unknown' it should be shortID, so we fall back to displaying the ID.